samples/wifi/sta/src/main.c

```diff
@@ -0,0 +1,257 @@
+		uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+		case NET_EVENT_WIFI_CONNECT_RESULT:
+			handle_wifi_connect_result(cb);
+			break;
+		case NET_EVENT_WIFI_DISCONNECT_RESULT:
+			handle_wifi_disconnect_result(cb);
+			break;
+		default:
+			break;
+	}
+}
+
+static void print_dhcp_ip(struct net_mgmt_event_callback *cb)
+{
+	/* Get DHCP info from struct net_if_dhcpv4 and print */
+	const struct net_if_dhcpv4 *dhcpv4 = cb->info;
+	const struct in_addr *addr = &dhcpv4->requested_ip;
+	char dhcp_info[128];
+
+	net_addr_ntop(AF_INET, addr, &dhcp_info, sizeof(dhcp_info));
+
+	printk("IP address: %s\n", dhcp_info);
+}
+
+static void net_mgmt_event_handler(struct net_mgmt_event_callback *cb,
+		uint32_t mgmt_event, struct net_if *iface)
+{
+	switch (mgmt_event) {
+		case NET_EVENT_IPV4_DHCP_BOUND:
+			print_dhcp_ip(cb);
+			break;
+		default:
+			break;
+	}
+}
+
+static int __wifi_args_to_params(struct wifi_connect_req_params *params)
+{
+	params->timeout = SYS_FOREVER_MS;
+
+	/* SSID */
+	params->ssid = CONFIG_STA_SAMPLE_SSID;
+	params->ssid_length = strlen(params->ssid);
+
+	params->security = CONFIG_STA_KEY_MGMT_SELECT;
+	if (params->security) {
+		params->psk = CONFIG_STA_SAMPLE_PWD;
+		params->psk_length = strlen(params->psk);
+	}
+	params->channel = WIFI_CHANNEL_ANY;
+
+	/* MFP (optional) */
+	params->mfp = WIFI_MFP_OPTIONAL;
+
+	return 0;
+}
+
+static int wifi_connect(void)
+{
+	struct net_if *iface = net_if_get_default();
+	static struct wifi_connect_req_params cnx_params;
+
+	__wifi_args_to_params(&cnx_params);
+
+	context.connecting = true;
+
+	if (net_mgmt(NET_REQUEST_WIFI_CONNECT, iface,
+				&cnx_params, sizeof(struct wifi_connect_req_params))) {
+		printk("Connection request failed\n");
+		context.connecting = false;
+
+		return -ENOEXEC;
+	}
+
+	printk("Connection requested\n");
+
+	return 0;
+}
+
+static int wifi_disconnect(void)
+{
+	struct net_if *iface = net_if_get_default();
+	int status;
+
+	context.disconnecting = true;
+
+	status = net_mgmt(NET_REQUEST_WIFI_DISCONNECT, iface, NULL, 0);
+
+	if (status) {
+		context.disconnecting = false;
+
+		if (status == -EALREADY) {
+			printk("Already disconnected\n");
+		} else {
+			printk("Disconnect request failed\n");
+			return -ENOEXEC;
+		}
+	} else {
+		printk("Disconnect requested\n");
+	}
+
+	return 0;
+}
+
+void main(void)
+{
+	context.all = 0U;
+
+	net_mgmt_init_event_callback(&wifi_shell_mgmt_cb,
+			wifi_mgmt_event_handler,
+			WIFI_SHELL_MGMT_EVENTS);
+
+	net_mgmt_add_event_callback(&wifi_shell_mgmt_cb);
+
+
+	net_mgmt_init_event_callback(&net_shell_mgmt_cb,
+			net_mgmt_event_handler,
+			NET_EVENT_IPV4_DHCP_BOUND);
+
+	net_mgmt_add_event_callback(&net_shell_mgmt_cb);
+
+	/* For now hardcode to 128MHz */
+	nrfx_clock_divider_set(NRF_CLOCK_DOMAIN_HFCLK,
+			NRF_CLOCK_HFCLK_DIV_1);
+	printk("Starting %s with CPU frequency: %d MHz\n", CONFIG_BOARD, SystemCoreClock/MHZ(1));
+	k_sleep(K_SECONDS(1));
+
+	wifi_connect();
+	k_sleep(K_SECONDS(30));
```

Comment:
Use a semaphore to block till its connected, just in case if it takes a bit longer.

Reply:
Within 30 sec, able to connect . So, used k_sleep().

Reply:
ok, lets leave this for now.